lib/puppet/pops/loader/puppet_function_instantiator.rb

```diff
@@ -43,7 +43,8 @@ def self.create(loader, typed_name, source_ref, pp_code_string)
     # Adapt the function definition with loader - this is used from logic contained in it body to find the
     # loader to use when making calls to the new function API. Such logic have a hard time finding the closure (where
     # the loader is known - hence this mechanism
-    Puppet::Pops::Adapters::LoaderAdapter.adapt(the_function_definition).loader = loader
+    private_loader = loader.private_loader
+    Puppet::Pops::Adapters::LoaderAdapter.adapt(the_function_definition).loader = private_loader
```

Comment:
This is not correct. We are currently not using the private loader as that is for functions that are declared to be private - and that means that they are private to the module. Until we have that, all functions in a module are public. All modules that are allowed to load public things from the module should be able to access things loaded from its public loader, but not from its private. The problem is elsewhere - in that a search should start from the private counterpart for any loaded item.